src/proof/decoded_proof_nodes.rs

```diff
@@ -0,0 +1,104 @@
+use crate::{nodes::TrieNode, proof::ProofNodes, HashMap, Nibbles};
+use alloy_primitives::Bytes;
+use alloy_rlp::Decodable;
+use core::ops::Deref;
+
+#[allow(unused_imports)]
```

Comment:
do we need this, I see Vec being used below

Reply:
whoops nope, is copypasta from `proof_nodes`